File: apps/demo-android/app/src/main/java/com/cinacoin/demo/ui/AuthScreen.kt
package com.cinacoin.demo.ui

import androidx.compose.foundation.layout.*
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.*
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.cinacoin.appkit.auth.AuthManager
import kotlinx.coroutines.launch

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun AuthScreen() {
    val currentUser by AuthManager.currentUser.collectAsState()
    val isAuthenticated = currentUser != null
    var email by remember { mutableStateOf("") }
    var password by remember { mutableStateOf("") }
    var isLoading by remember { mutableStateOf(false) }
    var error by remember { mutableStateOf<String?>(null) }
    val scope = rememberCoroutineScope()

    Column(
        modifier = Modifier
            .fillMaxSize()
            .padding(16.dp),
        verticalArrangement = Arrangement.spacedBy(16.dp)
    ) {
        // Status Card
        Card(modifier = Modifier.fillMaxWidth()) {
            Column(
                modifier = Modifier.padding(24.dp),
                horizontalAlignment = Alignment.CenterHorizontally
            ) {
                Icon(
                    imageVector = if (isAuthenticated) Icons.Default.Person else Icons.Default.PersonOff,
                    contentDescription = null,
                    modifier = Modifier.size(48.dp)
                )
                Spacer(Modifier.height(8.dp))
                Text(
                    text = if (isAuthenticated) "Authenticated" else "Not Authenticated",
                    fontWeight = FontWeight.SemiBold,
                    fontSize = 18.sp
                )
                currentUser?.let {
                    Text(it.email ?: it.userId, fontSize = 14.sp, color = MaterialTheme.colorScheme.onSurfaceVariant)
                }
            }
        }

        if (!isAuthenticated) {
            // Social Login
            Text("Social Login", fontWeight = FontWeight.Bold)
            OutlinedButton(
                onClick = { /* social login */ },
                modifier = Modifier.fillMaxWidth()
            ) { Icon(Icons.Default.Public, null); Spacer(Modifier.width(8.dp)); Text("Sign in with Google") }

            OutlinedButton(
                onClick = { /* social login */ },
                modifier = Modifier.fillMaxWidth()
            ) { Icon(Icons.Default.Code, null); Spacer(Modifier.width(8.dp)); Text("Sign in with GitHub") }

            OutlinedButton(
                onClick = { /* social login */ },
                modifier = Modifier.fillMaxWidth()
            ) { Icon(Icons.Default.Chat, null); Spacer(Modifier.width(8.dp)); Text("Sign in with Discord") }

            // Email Login
            Text("Email Login", fontWeight = FontWeight.Bold)
            OutlinedTextField(
                value = email,
                onValueChange = { email = it },
                label = { Text("Email") },
                modifier = Modifier.fillMaxWidth()
            )
            OutlinedTextField(
                value = password,
                onValueChange = { password = it },
                label = { Text("Password") },
                modifier = Modifier.fillMaxWidth()
            )
            Row(horizontalArrangement = Arrangement.spacedBy(12.dp)) {
                Button(
                    onClick = { /* login */ },
                    modifier = Modifier.weight(1f),
                    enabled = !isLoading
                ) { Text("Login") }
                OutlinedButton(
                    onClick = { /* register */ },
                    modifier = Modifier.weight(1f),
                    enabled = !isLoading
                ) { Text("Register") }
            }
        } else {
            Button(
                onClick = { AuthManager.signOut() },
                modifier = Modifier.fillMaxWidth(),
                colors = ButtonDefaults.buttonColors(containerColor = MaterialTheme.colorScheme.error)
            ) { Icon(Icons.Default.Logout, null); Spacer(Modifier.width(8.dp)); Text("Sign Out") }
        }

        error?.let { Text(it, color = MaterialTheme.colorScheme.error, fontSize = 12.sp) }
    }
}

File: apps/demo-flutter/lib/screens/auth_screen.dart
import 'package:flutter/material.dart';
import 'package:cinacoin/appkit_auth.dart';

class AuthScreen extends StatefulWidget {
  const AuthScreen({super.key});
  @override
  State<AuthScreen> createState() => _AuthScreenState();
}

class _AuthScreenState extends State<AuthScreen> {
  final _authManager = AuthManager();
  final _emailController = TextEditingController();
  final _passwordController = TextEditingController();
  bool _isLoading = false;
  String? _error;

  @override
  void dispose() {
    _emailController.dispose();
    _passwordController.dispose();
    super.dispose();
  }

  @override
  Widget build(BuildContext context) {
    final theme = Theme.of(context);
    final user = _authManager.currentUser;

    return ListView(
      padding: const EdgeInsets.all(16),
      children: [
        // Status
        Card(
          child: Padding(
            padding: const EdgeInsets.all(16),
            child: Column(
              children: [
                Icon(
                  user != null ? Icons.check_circle : Icons.person_off,
                  size: 48,
                  color: user != null ? theme.colorScheme.primary : theme.colorScheme.error,
                ),
                const SizedBox(height: 8),
                Text(
                  user != null ? 'Authenticated' : 'Not authenticated',
                  style: theme.textTheme.titleMedium,
                ),
                if (user != null) ...[
                  const SizedBox(height: 4),
                  Text(user.email ?? user.userId),
                ],
              ],
            ),
          ),
        ),
        const SizedBox(height: 24),

        // Social Login Buttons
        if (user == null) ...[
          Text('Social Login', style: theme.textTheme.titleMedium),
          const SizedBox(height: 12),
          _socialButton('Sign in with Google', Icons.g_mobiledata, () => _socialLogin('google')),
          const SizedBox(height: 8),
          _socialButton('Sign in with GitHub', Icons.code, () => _socialLogin('github')),
          const SizedBox(height: 8),
          _socialButton('Sign in with Discord', Icons.discord, () => _socialLogin('discord')),
          const SizedBox(height: 24),

          // Email Login
          Text('Email Login', style: theme.textTheme.titleMedium),
          const SizedBox(height: 12),
          TextField(
            controller: _emailController,
            decoration: const InputDecoration(
              labelText: 'Email',
              border: OutlineInputBorder(),
              prefixIcon: Icon(Icons.email),
            ),
            keyboardType: TextInputType.emailAddress,
          ),
          const SizedBox(height: 12),
          TextField(
            controller: _passwordController,
            decoration: const InputDecoration(
              labelText: 'Password',
              border: OutlineInputBorder(),
              prefixIcon: Icon(Icons.lock),
            ),
            obscureText: true,
          ),
          const SizedBox(height: 12),
          Row(
            children: [
              Expanded(
                child: ElevatedButton(
                  onPressed: _isLoading ? null : _login,
                  child: const Text('Login'),
                ),
              ),
              const SizedBox(width: 12),
              Expanded(
                child: OutlinedButton(
                  onPressed: _isLoading ? null : _register,
                  child: const Text('Register'),
                ),
              ),
            ],
          ),
        ] else ...[
          // Sign Out
          ElevatedButton.icon(
            onPressed: () {
              _authManager.signOut();
              setState(() {});
            },
            icon: const Icon(Icons.logout),
            label: const Text('Sign Out'),
            style: ElevatedButton.styleFrom(
              backgroundColor: theme.colorScheme.error,
              foregroundColor: Colors.white,
            ),
          ),
        ],

        if (_error != null) ...[
          const SizedBox(height: 16),
          Text(_error!, style: TextStyle(color: theme.colorScheme.error)),
        ],
      ],
    );
  }

  Widget _socialButton(String label, IconData icon, VoidCallback onTap) {
    return SizedBox(
      width: double.infinity,
      child: OutlinedButton.icon(
        onPressed: _isLoading ? null : onTap,
        icon: Icon(icon),
        label: Text(label),
        style: OutlinedButton.styleFrom(
          padding: const EdgeInsets.symmetric(vertical: 12),
        ),
      ),
    );
  }

  Future<void> _socialLogin(String provider) async {
    setState(() { _isLoading = true; _error = null; });
    try {
      switch (provider) {
        case 'google':
          await _authManager.signInWithGoogle();
        case 'github':
          await _authManager.signInWithGitHub();
        case 'discord':
          await _authManager.signInWithDiscord();
      }
    } catch (e) {
      setState(() => _error = e.toString());
    } finally {
      setState(() => _isLoading = false);
    }
  }

  Future<void> _login() async {
    setState(() { _isLoading = true; _error = null; });
    try {
      await _authManager.login(
        email: _emailController.text,
        password: _passwordController.text,
      );
    } catch (e) {
      setState(() => _error = e.toString());
    } finally {
      setState(() => _isLoading = false);
    }
  }

  Future<void> _register() async {
    setState(() { _isLoading = true; _error = null; });
    try {
      await _authManager.register(
        email: _emailController.text,
        password: _passwordController.text,
      );
    } catch (e) {
      setState(() => _error = e.toString());
    } finally {
      setState(() => _isLoading = false);
    }
  }
}
